DungeonWorld/DAO/DataAccesObject.cs

```diff
@@ -1,62 +1,26 @@
 ﻿using DungeonWorld.Model;
 using System;
 using System.Collections.Generic;
-using System.Data.Entity;
 using System.Linq;
-using System.Threading.Tasks;
 
 namespace DungeonWorld.DAO
 {
     public class DataAccesObject
     {
         public static DWModel context = new DWModel();
 
-        public static Race GetRaceById(int id)
-        {
-            
-            return context.Races.Where(r => r.IDRace == id).FirstOrDefault();
-            
+        public static Race GetRaceById(int id) => context.Races.FirstOrDefault(r => r.RaceId == id);
```

Comment:
To samo co wyżej, nie musisz robić `Where()`. `Where()` zwraca od razu listę, First/Single jeden obiekt. Nie ma co męczyć cpu niepotrzebnym kodem.

Druga sprawa, nigdzie nie sprawdzasz, czy w bazie rasa o takim `Id` występuje. Może warto by było zrobić jakieś sprawdzenie w kodzie?

```csharp
var race = context.Races.FirstOrDefault(r => r.RaceId == id) 
    ?? throw new ArgumentNullException(nameof(race));
```

Może lepiej w tym miejscu warto by było użyć `SingleOrDefault` jeśli spodziewasz się jednej rasy po tym `Id` bądź w ogóle użyj `Single()`? To powinno wynikać z twojej logiki, czy jesteś w stu procentach pewny, że rasa występuje? jeśli tak, daj `Single()` i jesli wywali ci wyjątek że nie ma elementu wtedy znaczy to, że coś nie działa tak jak powinno. Jeśli może być taki przypadek że rasy może nie być wtedy zwracanie nulla z `...OrDefault()` jest git. Tylko sprawdzaj w jakiś sposób, czy tego nulla nie dostajesz